src/Driver/UpgradedSocket.php

```diff
@@ -48,14 +51,14 @@ public function __destruct()
         $this->close();
     }
 
-    public function write(string $data): Future
+    public function write(string $data): void
     {
-        return $this->socket->write($data);
+        $this->socket->write($data);
     }
 
-    public function end(string $finalData = ""): Future
+    public function end(string $finalData = ""): void
```

Comment:
The optional argument has been removed.

Reply:
I wasn't sure whether I should be changing the `Future`s to `void`s anywhere else - only hit this one because the type changed in Socket.